### PR TITLE
Demote `style-servo` to only run in the "stable" set.

### PR DESCRIPTION
**Pull request by nnethercote**
It's the longest-running benchmark in the suite, but doesn't provide results that are all that different to other benchmarks. Plus its importance in the Rust ecosystem is less than it once was. But we keep it as part of the "stable" set to preserve data continuity.

Best reviewed one commit at a time.

### CODE REVIEW

File: collector/src/execute.rs [hunk 1]
```diff
@@ -131,7 +131,11 @@ struct BenchmarkConfig {
     #[serde(default)]
     touch_file: Option<String>,
 
-    category: Category,
+    /// If this field is `None` when read from `perf-config.json`:
+    /// - If `supports_stable` is true, it will be replaced with
+    ///   `Some(Primary)` when the result is recorded in the DB.
+    /// - If `supports_stable` is false, an error will be returned.
+    category: Option<Category>,
```
Comment (Mark-Simulacrum) [inline collector/src/execute.rs:138]: Maybe we should have a StableOnly category or something like that? (Potentially in the long run fully replacing supports_stable with that category).

**Comment by Mark-Simulacrum** [comment]
I'm not sure I'm a fan of omitting the category getting you an implicit behavior of not running at all, seems unnecessary and rather annoying to understand/debug if you don't know about the stable-only support.

I think we talked on Zulip about needing to disable the master point on the [dashboard](perf.rust-lang.org/dashboard.html) before we do this, right? Since now style-servo will no longer run there?

**Comment by nnethercote** [comment]
> I'm not sure I'm a fan of omitting the category getting you an implicit behavior of not running at all, seems unnecessary and rather annoying to understand/debug if you don't know about the stable-only support.

I agree! It would be better if "stable" was a distinct category from "primary" and "secondary" and every benchmark belonged to exactly one category, and "supports_stable" went away. Maybe we will end up there, but there needs to be a transition period. And I didn't want to mess with the database format for now. This PR is the minimal thing that will work, I think.

I chose to disable style-servo first because that will give us immediate headroom, in terms of running time, for any increases in benchmark sizes, and periods of overlap where both the old and new versions of benchmarks are live.

> I think we talked on Zulip about needing to disable the master point on the [dashboard](perf.rust-lang.org/dashboard.html) before we do this, right? Since now style-servo will no longer run there?

If it uses the `bench_published` command, then style-servo *will* run. That's the only place it will run.

**Comment by Mark-Simulacrum** [comment]
> If it uses the bench_published command, then style-servo will run. That's the only place it will run.

The master point doesn't though? The dashboard just pulls from the latest master commit benchmarked and assumes the supports stable benchmarks will be there. It should be pretty easy to drop it -- just deleting [these lines](https://github.com/rust-lang/rustc-perf/blob/master/site/src/request_handlers/dashboard.rs#L74-L76) I think -- but I think that should probably happen here. It may also be worth a comment on that page of some kind (...this does not contain master intentionally...).

**Comment by Mark-Simulacrum** [comment]
Modulo the dashboard question, I think this looks good to me.

**Comment by nnethercote** [comment]
I have completely changed how I'm handling stable benchmarks, I think the new approach is much better. And I've added the master removal. The first commit is unchanged from yesterday but the latter three commits are very different and will require re-review, thanks!

**Pull request merged**